language-support/ts/codegen/src/Main.hs

```diff
@@ -123,7 +129,19 @@ genDefDataType curModName tpls def = case unTypeConName (dataTypeCon def) of
     [conName] -> case dataCons def of
         DataSynonym{} -> ((makeType ["unknown;"], makeSer ["jtv.unknownJson,"]), Set.empty)  -- TODO(NICK)
         DataVariant{} -> ((makeType ["unknown;"], makeSer ["jtv.unknownJson,"]), Set.empty)  -- TODO(MH): make variants type safe
-        DataEnum{} -> ((makeType ["unknown;"], makeSer ["jtv.unknownJson,"]), Set.empty)  -- TODO(MH): make enum types type safe
+        DataEnum enumCons ->
+          let
+            typeDesc =
+                [ "export enum " <> conName <> "{"] ++
+                [ "  " <> cons <> " = " <> "\'" <> cons <> "\'" <> ","
+                | (VariantConName cons) <- enumCons] ++
+                [ "}" ]
+            serDesc =
+                ["  () => jtv.oneOf("] ++
+                ["    jtv.constant(" <> conName <> "." <> cons <> ")," | VariantConName cons <- enumCons] ++
```

Comment:
This line is tricky. On first reading, I thought it's wrong because I confused the two object languages. That is clearly not the case and it's alright. Generating code for language `X` from language `Y` in language `Z` is a tough best. 😃